app/Helpers/database/code-note.php

```diff
@@ -82,7 +82,7 @@ function submitCodeNote2(string $username, int $gameID, int $address, string $no
     if (
         $i !== false
         && $permissions <= Permissions::JuniorDeveloper
-        && $currentNotes[$i]['User'] !== $user
+        && $currentNotes[$i]['User'] !== $user->User
```

Comment:
@Jamiras I believe your find is the result of a bug in this conditional that emerged in the V3 branch. Beforehand, I think `$user` was a string containing the username, now it's an object. Should be fixed now.

Reply:
I agree with this assessment.